File: Snooper/src/test/java/com/prateekj/snooper/formatter/ResponseFormatterFactoryTest.java
package com.prateekj.snooper.formatter;

import org.junit.Before;
import org.junit.Test;

import static junit.framework.Assert.assertTrue;

public class ResponseFormatterFactoryTest {

  private ResponseFormatterFactory factory;

  @Before
  public void setUp() throws Exception {
    factory = new ResponseFormatterFactory();
  }

  @Test
  public void shouldReturnXmlFormatter() throws Exception {
    assertTrue(factory.getFor("xml") instanceof XmlFormatter);
  }

  @Test
  public void shouldReturnJsonFormatter() throws Exception {
    assertTrue(factory.getFor("json") instanceof JsonResponseFormatter);
  }

  @Test
  public void shouldReturnPlainTextFormatter() throws Exception {
    assertTrue(factory.getFor("plain") instanceof PlainTextFormatter);
  }
}

File: Snooper/src/test/java/com/prateekj/snooper/formatter/ResponseFormatterFactoryTest.kt
package com.prateekj.snooper.formatter

import org.junit.Before
import org.junit.Test

import junit.framework.Assert.assertTrue

class ResponseFormatterFactoryTest {

  private lateinit var factory: ResponseFormatterFactory

  @Before
  @Throws(Exception::class)
  fun setUp() {
    factory = ResponseFormatterFactory()
  }

  @Test
  @Throws(Exception::class)
  fun shouldReturnXmlFormatter() {
    assertTrue(factory.getFor("xml") is XmlFormatter)
  }

  @Test
  @Throws(Exception::class)
  fun shouldReturnJsonFormatter() {
    assertTrue(factory.getFor("json") is JsonResponseFormatter)
  }

  @Test
  @Throws(Exception::class)
  fun shouldReturnPlainTextFormatter() {
    assertTrue(factory.getFor("plain") is PlainTextFormatter)
  }
}